### PR TITLE
python 3.8.18; re-enable setuptools downstream test

### PR DESCRIPTION
**Pull request by h-vetinari**
As [requested](https://github.com/conda-forge/python-feedstock/pull/604#issuecomment-1404947149) by @isuruf

### CODE REVIEW

**Comment by mbargull** [comment]
Thanks for reverting this.
We don't need to put out additional builds for this again, though (i.e., not increment the build number).
(I'd also be fine to put a `[[ci skip] [skip ci] ***NO_CI***` in the commit message to avoid the build altogether.)

**Comment by h-vetinari** [comment]
> @mbargull: Thanks for reverting this.

No problem - though I don't understand how this is going to help, given that you said in #604:
> @mbargull: Downstream test issues are non-functional due to [conda/conda-build#4750](https://github.com/conda/conda-build/issues/4750) (i.e., an issue we experience through `conda-mambabuild` caused by a `conda-build`+`libsolv=0.7.23` incompatibility).

And indeed, travis has failed already...

> @mbargull: We don't need to put out additional builds for this again, though (i.e., not increment the build number).

That was my assumption as well, but @jakirkham had asked me specifically in #618 to bump:
> @jakirkham: Can we please bump the `build/number` as well? Changing `test` affects the packages. Especially when passing to `conda build --test <pkg>`

**Comment by h-vetinari** [comment]
@isuruf @mbargull 
Now that the fix in conda-build has been merged & released, this PR is passing again.

**Review by varlackc — APPROVED** [review]
Looks Good To Me

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
```
         C:\Miniforge\Library\usr\bin\git.exe name-rev --name-only HEAD > "D:\bld\python-split_1692915799848\work\PCbuild\obj\38amd64_Release\pythoncore\gitbranch.txt"
         fatal: Not a git repository (or any parent up to mount point /cygdrive/d)
         Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

**Comment by h-vetinari** [comment]
Hah, funny timing, a new python 3.8 just got [released](https://discuss.python.org/t/python-3-11-5-3-10-13-3-9-18-and-3-8-18-is-now-available/32254)

**Comment by h-vetinari** [comment]
I don't know why the other branches pass while this one fails. I looked at the diff between this and the `3.9` branch and couldn't find something that looked git-related. Perhaps some interaction between a new git version and old expectations in the upstream windows build scripts.

**Comment by h-vetinari** [comment]
Same error persists:
```
       _GetBuildInfo:
         Getting build info from C:\Miniforge\Library\usr\bin\git.exe
         Creating directory "D:\bld\python-split_1695116942199\work\PCbuild\obj\38amd64_Release\pythoncore\".
         C:\Miniforge\Library\usr\bin\git.exe name-rev --name-only HEAD > "D:\bld\python-split_1695116942199\work\PCbuild\obj\38amd64_Release\pythoncore\gitbranch.txt"
         fatal: Not a git repository (or any parent up to mount point /cygdrive/d)
         Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

**Comment by xhochy** [comment]
@h-vetinari I think the following error is actually the one why it is failing:

```
2023-09-19T09:53:19.4955837Z        "D:\bld\python-split_1695116942199\work\PCbuild\pcbuild.proj" (Build target) (1) ->
2023-09-19T09:53:19.4956846Z        "D:\bld\python-split_1695116942199\work\PCbuild\pythoncore.vcxproj" (Build target) (2) ->
2023-09-19T09:53:19.4957496Z        (PrepareForBuild target) -> 
2023-09-19T09:53:19.4958758Z          C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(456,5): error MSB8020: The build tools for Visual Studio 2015 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install Visual Studio 2015 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [D:\bld\python-split_1695116942199\work\PCbuild\pythoncore.vcxproj]
```

**Comment by xhochy** [comment]
@h-vetinari Would it be OK for me to rerender here and pin the azure image back to `windows-2019` manually?

File: .azure-pipelines/azure-pipelines-win.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 jobs:
 - job: win
   pool:
-    vmImage: windows-2022
+    vmImage: windows-2019
```
Comment (h-vetinari) [inline .azure-pipelines/azure-pipelines-win.yml:8]: There's a way to make this persist across rerenders, if you add the following to the cbc:
```

azure:
  settings_win:
    pool:
      vmImage: windows-2019
```

**Pull request merged**